### ISSUE TITLE
Implementation: upload_waveform for TablePulseTemplate and SequencePulseTemplate

### ISSUE THREAD

**Issue by lumip** [qutech/qc-toolkit#24]
Issue #20 resulted in a (prototype of a) *PulseHardwareUploadInterface* which enables the implementation of the *upload_waveform(..)* method of the *PulseTemplate* class hierarchy.
Implement and test *upload_waveform(..)* for *TablePulseTemplate*, *SequencePulseTemplate*.

Part of #13.

**Comment by lumip** [comment]
Afflicted by #27 . This issue then refers to the implementation of build_sequence().

**Comment by lumip** [comment]
*build_sequence* and *requires_stop* have been implemented for SequencePulseTemplate and TablePulseTemplate.
For SequencePulseTemplate, the parameter mapping has to be regarded. I encapsulated this in a private *__map_parameters* method which is called from *build_sequence* and has yet to be implemented. 
@j340m3 Since you know more about how the mapping works in detail, I'll leave that to you, as we have discussed.

Tests will follow in the respective issues #30 and #32 .
